.ci/build_demo.sh

```diff
@@ -0,0 +1,150 @@
+    echo "A network $ELASTICSEARCH_NETWORK has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The network $ELASTICSEARCH_NETWORK already exists."
+fi
+
+if [[ -z "$(docker ps -q -f name="$ELASTICSEARCH_CONTAINER")" ]]; then
+    echo "Creating a shared elasticsearch container $ELASTICSEARCH_CONTAINER..."
+    docker run --name "$ELASTICSEARCH_CONTAINER" --ulimit "nofile=65536:65536" --env "discovery.type=single-node" -d "docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.2"
+    echo "The elasticsearch container has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The elasticsearch container $ELASTICSEARCH_CONTAINER already exists."
+fi
+
+if [[ "$NETWORK_CONNECT" == 'true' ]]; then
+    echo "Connecting the container to the network...."
+    docker network connect "$ELASTICSEARCH_NETWORK" "$ELASTICSEARCH_CONTAINER"
+fi
+
+ELASTICSEARCH_IP=$(docker inspect -f "{{ with (index .NetworkSettings.Networks \"$ELASTICSEARCH_NETWORK\") }}{{ .IPAddress }}{{ end }}" "$ELASTICSEARCH_CONTAINER")
+if [[ -z "$ELASTICSEARCH_IP" ]]; then
+    echo -e "${RED}The IP address of the elasticsearch container inside the shared network could not be obtained!${NC}"
+    exit 1
+else
+    echo -e "The IP address of ${BLUE}$ELASTICSEARCH_CONTAINER${NC} inside ${BLUE}$ELASTICSEARCH_NETWORK${NC} is ${BLUE}$ELASTICSEARCH_IP${NC}"
+fi
+
+# Copy the demo Docker configuration
+cp docker/conf/docker-compose.demo.yml.dist docker-compose.yml
+
+echo -e "${BLUE}Building Docker images and starting up the containers...${NC}"
+
+# Export variables used in Docker Compose configuration
+export DEMO_NAME
+export ELASTICSEARCH_NETWORK
+export ELASTICSEARCH_IP
+
+# Build the Docker image and start the containers
+docker-compose up --build --force-recreate -d
+
+# Configure the nginx
+WEBSERVER_PORT=$(docker-compose port webserver 8080)
+echo "upstream $JOB_NAME-upstream {
+    server 127.0.0.1:${WEBSERVER_PORT##*:};
+}
+
+server {
+    listen 80;
+    server_name $DEMO_NAME.$DEMO_DOMAIN *.$DEMO_NAME.$DEMO_DOMAIN;
+
+    location / {
+        proxy_pass http://$JOB_NAME-upstream;
+        proxy_redirect off;
+        proxy_set_header Host \$host;
+        proxy_set_header X-Real-IP \$remote_addr;
+        proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host \$host;
+    }
+}" > "/etc/nginx/conf.d/$JOB_NAME.conf"
+
+# Reload nginx to apply the new configuration
+# "jenkins" user has been allowed to run "nginx" command as super-user without password prompt via /etc/sudoers configuration
+# see https://www.digitalocean.com/community/tutorials/how-to-edit-the-sudoers-file-on-ubuntu-and-centos#how-to-modify-the-sudoers-file
+sudo nginx -s reload
+
+echo -e "${BLUE}Building the application...${NC}"
+
+# Wait a moment for postgres to start, ten seconds should suffice
+# (avoids having to implement a check according to https://docs.docker.com/compose/startup-order/)
+sleep 10
```

Comment:
It doesn't wait for them until they are ready, but just until they are started. The postgres container reboots itself right after its created (I think it's because it needs to switch users). There were some problems with it and they were completely erradicated by this short wait.

Reply:
just some hints from web https://gist.github.com/nicerobot/1136dcfba6ce3da67ce3ded5101a4078